pandas/_libs/hashing.pyx

```diff
@@ -15,7 +15,7 @@ DEF dROUNDS = 4
 
 
 @cython.boundscheck(False)
-def hash_object_array(object[:] arr, object key, object encoding='utf8'):
+def hash_object_array(ndarray[object] arr, object key, object encoding='utf8'):
```

Comment:
I get the impression that memory views used in tandem with `object` will throw warnings cc @jbrockmendel

Reply:
that makes sense.  this would also raise ATM if we were to pass a read-only ndarray[object], so a worthwhile change